### PR TITLE
Configure se050 backend on boot

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Depends on:

- https://github.com/Nitrokey/trussed-se050-backend/pull/31
- https://github.com/trussed-dev/trussed/pull/165

### CODE REVIEW

**Review by robin-nitrokey — COMMENTED** [review]
Can we get rid of the `cfg` attributes by always passing the service even if it is not needed for the non-se050 case?

**Comment by sosthene-nitrokey** [comment]
This is actually difficult to do. The `Service` used by usbip is `pub struct Service<S: StoreProvider, D: Dispatch>(Rc<RefCell<trussed::Service<Platform<S>, D>>>);`, so there is no sound way to have a `&mut trussed::Service` at the same time as the `ClientBuilder`.

Maybe we need to change the `make_client` to be a trait implementation with a `build` method, and also a method to get a mut access to the `trussed::Service`.

That requires also updating the pc_usbip_runner crate.

**Comment by sosthene-nitrokey** [comment]
I think i'll just put it behind an option and pass `None` in the usbip runner, with an unwrap.
Worst case in the future if an new update expects it to always be available, it only breaks the usbip runner.

**Comment by robin-nitrokey** [comment]
We can get rid of the option once https://github.com/Nitrokey/nitrokey-3-firmware/pull/505 is merged.

**Review by robin-nitrokey — COMMENTED** [review]
LGTM.  We should use upstream Trussed instead of your fork though.

File: Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ cbor-smol = { git = "https://github.com/Nitrokey/cbor-smol.git", tag = "v0.4.0-n
 fido-authenticator = { git = "https://github.com/Nitrokey/fido-authenticator.git", tag = "v0.1.1-nitrokey.15" }
 lpc55-hal = { git = "https://github.com/Nitrokey/lpc55-hal", tag = "v0.3.0-nitrokey.2" }
 serde-indexed = { git = "https://github.com/nitrokey/serde-indexed.git", tag = "v0.1.0-nitrokey.2" }
-trussed = { git = "https://github.com/Nitrokey/trussed.git", tag = "v0.1.0-nitrokey.19" }
+trussed = { git = "https://github.com/sosthene-nitrokey/trussed.git", rev = "42db356db6d457a6bdb78fba8a1521c91887eb0a" }
```
Comment (robin-nitrokey) [inline Cargo.toml:25]: should use upstream Trussed or a tag

File: Cargo.toml [hunk 2]
```diff
@@ -48,7 +48,7 @@ trussed-hkdf = { git = "https://github.com/trussed-dev/trussed-staging.git", tag
 trussed-rsa-alloc = { git = "https://github.com/trussed-dev/trussed-rsa-backend.git", rev = "9732a9a3e98af72112286afdc9b7174c66c2869a" }
 trussed-usbip = { git = "https://github.com/Nitrokey/pc-usbip-runner.git", tag = "v0.0.1-nitrokey.4" }
 trussed-se050-manage = { git = "https://github.com/Nitrokey/trussed-se050-backend.git", tag = "se050-manage-v0.1.0" }
-trussed-se050-backend = { git = "https://github.com/Nitrokey/trussed-se050-backend.git", rev = "23d3511276176da396b6c3e788cd1c2f4dd37c9d" }
+trussed-se050-backend = { git = "https://github.com/Nitrokey/trussed-se050-backend.git", rev = "e239875c32691ba36927a2a3652126958b804a01" }
```
Comment (robin-nitrokey) [inline Cargo.toml:51]: tag a release?

**Review by robin-nitrokey — APPROVED** [review]

**Review by daringer — APPROVED** [review]

**Pull request merged**